### PR TITLE
First implementation of revocation check support

### PR DESCRIPTION
**Pull request by nicolatimeus**
Signed-off-by: Nicola Timeus <nicola.timeus@eurotech.com>

* Adds support for enabling revocation check in `SSLManagerService` and `Jetty`.
* Added support for downloading and caching CRLs to `FileSystemKeystoreServiceImpl`.

### CODE REVIEW

File: kura/org.eclipse.kura.core.keystore/OSGI-INF/metatype/org.eclipse.kura.core.keystore.FilesystemKeystoreServiceImpl.xml [hunk 1]
```diff
@@ -45,8 +45,88 @@
             description="Specifies if the defined password will be randomized at the next keystore access. If this value is set to true and the keystore can be accessed, the password will be randomized and this field will automatically set to false.">
         </AD>
 
+        <AD id="crl.management.enabled"
```
Comment (MMaiero) [inline kura/org.eclipse.kura.core.keystore/OSGI-INF/metatype/org.eclipse.kura.core.keystore.FilesystemKeystoreServiceImpl.xml:48]: Maybe we need to expand the description. As a user I'm not sure what this option does

File: kura/org.eclipse.kura.core.keystore/OSGI-INF/metatype/org.eclipse.kura.core.keystore.FilesystemKeystoreServiceImpl.xml [hunk 2]
```diff
@@ -45,8 +45,88 @@
...
+            description="Enables or disables CRL cache.">
+        </AD>
+
+        <AD id="crl.update.interval"
+            name="CRL Force Update Interval"
+            type="Long"
+            cardinality="0"
+            required="true"
+            default="1"
+            description="Defines a time interval for forcing the update of cached CRLs, this interval will be considered in addition to CRL next update date.">
```
Comment (MMaiero) [inline kura/org.eclipse.kura.core.keystore/OSGI-INF/metatype/org.eclipse.kura.core.keystore.FilesystemKeystoreServiceImpl.xml:72]: Does this work only if a URL is set in `crl.urls`?

File: kura/org.eclipse.kura.core.keystore/OSGI-INF/metatype/org.eclipse.kura.core.keystore.FilesystemKeystoreServiceImpl.xml [hunk 3]
```diff
@@ -45,8 +45,88 @@
...
+            <Option label="HOURS" value="HOURS" />
+            <Option label="DAYS" value="DAYS" />
+        </AD>
+
+        <AD id="crl.store.path"
+            name="CRL Store Path"
+            type="String"
+            cardinality="0"
+            required="false"
+            default=""
```
Comment (MMaiero) [inline kura/org.eclipse.kura.core.keystore/OSGI-INF/metatype/org.eclipse.kura.core.keystore.FilesystemKeystoreServiceImpl.xml:115]: Does this require a full path or something like `file:<some path>`?

File: kura/org.eclipse.kura.core.keystore/OSGI-INF/metatype/org.eclipse.kura.core.keystore.FilesystemKeystoreServiceImpl.xml [hunk 4]
```diff
@@ -45,8 +45,88 @@
...
+        <AD id="verify.crl"
+            name="Enable CRL Verification"
+            type="Boolean"
+            cardinality="0"
+            required="true"
+            default="true"
+            description="If set to true, the downloaded CRLs will be stored only if signed with the public key of one of trusted certificates in this keystore.">
+        </AD>
+
+        <AD id="crl.urls"
```
Comment (MMaiero) [inline kura/org.eclipse.kura.core.keystore/OSGI-INF/metatype/org.eclipse.kura.core.keystore.FilesystemKeystoreServiceImpl.xml]: If URLs may be needed for the feature to work, maybe the field should be put on top after the enabling

File: kura/org.eclipse.kura.core/OSGI-INF/metatype/org.eclipse.kura.ssl.SslManagerService.xml [hunk 5]
```diff
@@ -52,6 +52,34 @@
...
             default=""
             description="Comma-separated list of allosed ciphers. If not specifed, all Java VM ciphers will be allowed."/>
+        
+         <AD id="ssl.revocation.check.enabled"
+            name="Revocation Check Enabled"
+            type="Boolean"
+            cardinality="0"
+            required="false"
+            default="false"
+            description="Specifies whether to check revocation status of server certificates."/>
```
Comment (MMaiero) [inline kura/org.eclipse.kura.core/OSGI-INF/metatype/org.eclipse.kura.ssl.SslManagerService.xml]: Maybe also here we need to enrich the description of the feature. Also unrelated, but I have noticed that there is an error here: "Comma-separated list of allo**w**ed ciphers"

File: kura/org.eclipse.kura.core.keystore/OSGI-INF/metatype/org.eclipse.kura.core.keystore.FilesystemKeystoreServiceImpl.xml (hunk 2 above)
Comment (nicolatimeus) [reply kura/org.eclipse.kura.core.keystore/OSGI-INF/metatype/org.eclipse.kura.core.keystore.FilesystemKeystoreServiceImpl.xml:72]: No, the HTTP distribution points specified in the trusted certificates in the keystore are considered as well, in addition to the URLs specified in configuration.

**Review by MMaiero — APPROVED** [review]

**Pull request merged**